### PR TITLE
Podcast player service and fragment

### PR DESCRIPTION
**Pull request by kkong006**
Change Log:
- Add PlayerService to play audio from a source in the background
- Add callback in HomeActivity to listen to changes in PlayerService
- Connect PlayerFragment play, pause, and seekbar to PlayerService
- Add button to ChannelFragment to play sample Episode

### CODE REVIEW

**Review by lepatrick714 — COMMENTED** [review]
Fuck give us 10 days to review

**Comment by lepatrick714** [comment]
Please update your branch with the current master branch.

File: app/src/main/java/appricottsoftware/clarity/HomeActivity.java [hunk 1]
```diff
@@ -45,11 +64,20 @@
     private static SettingFragment settingFragment;
     private static PlayerFragment playerFragment;
 
+    private Intent playerServiceIntent;
+    private MediaBrowserCompat mediaBrowser;
+    private MediaBrowserCompat.ConnectionCallback connectionCallback;
+    private MediaControllerCompat.Callback controllerCallback;
+    private MediaControllerCompat mediaController;
+
+    private Context context;
```
Comment (ttony678) [inline app/src/main/java/appricottsoftware/clarity/HomeActivity.java:73]: Why get context when you can use 'this'

**Comment by ttony678** [comment]
If the player is expanded and we press the back button, the player should collapse rather than closing the entire app, no?

**Review by ttony678 — DISMISSED** [review]
Check comments. I believe you can get that fixed soon so I'll approve the merge in the meantime.

File: app/src/main/java/appricottsoftware/clarity/HomeActivity.java (hunk 1 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/HomeActivity.java:73]: Runnable is an inner class that can't access 'this'.

**Comment by kkong006** [comment]
@ttony678 Replied

**Review by ttony678 — APPROVED** [review]
Thank you for addressing the back button issue

**Pull request merged**